packages/server-wallet/src/models/objective.ts

```diff
@@ -91,8 +91,8 @@ export class ObjectiveModel extends Model {
   readonly status!: DBObjective['status'];
   readonly type!: DBObjective['type'];
   readonly data!: DBObjective['data'];
-  createdAt!: string;
-  progressLastMadeAt!: string;
+  createdAt!: Date;
```

Comment:
I'm a bit surprised about this change. The DB schema hasn't changed. This seems like it's only changing the _type_ of `createdAt`. This, combined with changing the initial value of `createdAt` from `new Date().toISOString()` to `new Date(Date.now())` _that gets inserted_, seems to have changed the type of `createdAt` _that is returned_ when a row is fetched.

Can you explain this?
- Does switching from `new Date().toISOString()` to `new Date(Date.now())` change the data that gets stored?
- Or, is the data that gets stored the same, but changing the type of `createdAt` to `Date` change the data that gets returned?

Reply:
> > Objective timestamps are a mix of strings and `Date`s
> 
> It is the case that, as of this change, timestamps are _not_ a mix of strings and `Date`s, right? That is, timestamps are now `Date`s?

That's correct.

Reply:
@andrewgordstewart firstly, we don't need to pass `Date.now()` into the `Date` constructor, so I removed that in d192b2c.

You are correct to say the schema has not changed. This PR changes the data type passed into the query builder, and as a result the stored data has *not* changed, while the data returned *has* changed. This might seem surprising, but can be explained as follows:
1. During testing with the very same db schema, denoting the `createdAt` column as a timestamp, we can successfully insert either a `Date` *or* a `Date.toISOString()`. This implies that the type will be converted as necessary by knex before it gets to the DB driver.
2. Inside `ObjectiveModel.insert()` we return the result of the query *without* doing `.returning('*')`**. This means that you get out whatever you passed in. Previously this was a string but now it will be a `Date`, and this is bypassing any type conversion done by knex. 

So you pretty much had it right in your second bullet point. 

** we could do this, and maybe we should do this. It seems unnecessary because we construct the entire row before inserting it, so we "already have it". But, in future we might (say) add the timestamps in a `$beforeInsert` hook, and then we don't have the full row in scope at this point. There is also this question of datatypes that becomes harder to understand because the types can change when the return from the DB. So therefore 47ea69e.

Reply:
Thanks. A derivative of this explanation probably belongs in the repository, (eg. See [here](https://mtlynch.io/code-review-love/#4-answer-questions-with-the-code-itself)) since this is an important nuance that explains why the code behaves the way it does.

Reply:
The link you referenced suggests that the best solution in a case like this, is to refactor the code to eliminate confusion: which I think we effectively have already done.

In my opinion:

- the end state of the code works as you would expect
- we only needed an explanation here because the start state of the code was behaving in an unexpected way, which made the transition to the end state also quite confusing.
- the additions to the test reduce confusion by asserting on the return type
- 47ea69e adds an explanatory comment

I think this is sufficient, and I am not sure what else we could add? Happy to take suggestions, though.